src/SMTP.php

```diff
@@ -905,7 +914,15 @@ protected function parseHelloFields($type)
      * $from. Returns true if successful or false otherwise. If True
      * the mail transaction is started and then one or more recipient
      * commands may be called followed by a data command.
-     * Implements RFC 821: MAIL <SP> FROM:<reverse-path> <CRLF>.
+     * Implements RFC 821: MAIL <SP> FROM:<reverse-path> <CRLF> and
+     * two extensions, namely XVERP and SMTPUTF8.
+     *
+     * The server's EHLO response is not checked. If use of either
```

Comment:
I mentioned this earlier. You can check if the mail server supports an extension by looking in the `SMTP::$server_caps` array after EHLO has happened, like this:

    array_key_exists('SMTPUTF8', $this->server_caps)

Reply:
The check is easy, the next line is the question. I considered throwing an exception, but decided against it for the moment.
 - I didn't want to have SMTPUTF8 behave differently from XVERP in that function
 - Changing XVERP would change existing functionality (that noone should depend on, but...)
 - Callers can disable exceptions anyway

I could of course call setError(), like how sendCommand() would do when the `MAIL FROM` fails. Shall I?